lighthouse-core/runner.js

```diff
@@ -59,27 +46,49 @@ class Runner {
         data: sentryContext && sentryContext.extra,
       });
 
-      // If the URL isn't https and is also not localhost complain to the user.
-      if (parsedURL.protocol !== 'https:' && parsedURL.hostname !== 'localhost') {
-        log.warn('Lighthouse', 'The URL provided should be on HTTPS');
-        log.warn('Lighthouse', 'Performance stats will be skewed redirecting from HTTP to HTTPS.');
-      }
-
-      // canonicalize URL with any trailing slashes neccessary
-      const requestedUrl = parsedURL.href;
-
       // User can run -G solo, -A solo, or -GA together
       // -G and -A will run partial lighthouse pipelines,
       // and -GA will run everything plus save artifacts to disk
 
       // Gather phase
       // Either load saved artifacts off disk or from the browser
       let artifacts;
+      let requestedUrl;
       if (settings.auditMode && !settings.gatherMode) {
         // No browser required, just load the artifacts from disk.
         const path = Runner._getArtifactsPath(settings);
         artifacts = await assetSaver.loadArtifacts(path);
+        requestedUrl = artifacts.URL.requestedUrl;
+
+        if (!requestedUrl) {
+          throw new Error('Cannot run audit mode on empty URL');
+        }
+        if (opts.url && opts.url !== requestedUrl) {
+          throw new Error('Cannot run audit mode on different URL');
+        }
       } else {
+        // save the requestedUrl provided by the user
+        const rawRequestedUrl = opts.url;
+        if (typeof rawRequestedUrl !== 'string' || rawRequestedUrl.length === 0) {
+          throw new Error(`You must provide a url to the runner. '${rawRequestedUrl}' provided.`);
+        }
+
+        let parsedURL;
+        try {
+          parsedURL = new URL(opts.url);
+        } catch (e) {
+          throw new Error('The url provided should have a proper protocol and hostname.');
+        }
+
+        // If the URL isn't https and is also not localhost complain to the user.
+        if (parsedURL.protocol !== 'https:' && parsedURL.hostname !== 'localhost') {
```

Comment:
I _was_ gonna say two things...

1. we could use `is-on-https`'s `isSecureRecord` method here..
1. lets merge the strings and just say `'The URL provided should be on HTTPS. Performance stats might be skewed redirecting to HTTPS.'`

but tbh.. 

how about we nuke this check? Our redirects audit already catches perf cost of redirects. and if someone wants to audit a http page, they should be able to.

Reply:
also, in the future, i'd like to support `example.com` and just toss `https://` in front of it. psi supports this. i think wpt, too.

Reply:
yeah sg :)

Reply:
a bigger issue is service workers and how do you notice you aren't getting one because you typed in `http`, but agree there could be a better flow (we have top level warnings for that kind of thing now)